libs/sim/simulator.rs

```diff
@@ -53,7 +53,7 @@ impl<'a> Simulator {
         }
     }
 
-    pub fn query_storage(&mut self, storage_id: &str) -> &str {
+    pub fn query_storage(&mut self, storage_id: &str) -> String {
```

Comment:
This does not seem right. Why would storage return a copy instead of a reference?

Reply:
Because data as a string is owned by storage, so it is fine for storage to return a reference.  But if you return a reference to a value created in the aggregation function, that memory could disappear by the time you read it in main, because the aggregation function is over.

Reply:
This seems like a design flaw. The result that is computed by the UDF should be owned by storage after. It should not be bound to the life time of the UDF call.

Reply:
Okay, I switched it so every time we query storage we make the data of the storage node equal to the result of running the return function.  This means all sim_elements have to have a as_any_mut function, hence why there's a lot of changes.